sdks/python/apache_beam/io/gcp/gcsio.py

```diff
@@ -87,6 +87,39 @@
 MAX_BATCH_OPERATION_SIZE = 100
 
 
+def proxy_info_from_environment_var(proxy_env_var):
+  """Reads proxy info from the environment and converts to httplib2.ProxyInfo.
+  Args:
+    proxy_env_var: environment variable string to read, http_proxy or
+       https_proxy (in lower case).
+       Example: http://myproxy.domain.com:8080
+  Returns:
+    httplib2.ProxyInfo constructed from the environment string.
+  """
+  proxy_url = os.environ.get(proxy_env_var)
+  if not proxy_url or not proxy_env_var.lower().startswith('http'):
+    return httplib2.ProxyInfo(httplib2.socks.PROXY_TYPE_HTTP, None, 0)
+    logging.warn("Ignoring proxy_env_var, incorrect format")
+  proxy_protocol = proxy_env_var.lower().split('_')[0]
+  if not proxy_url.lower().startswith('http'):
+    logging.warn("proxy_info_from_url requires a protocol, which is always http or https.")
+    proxy_url = proxy_protocol + '://' + proxy_url
+  return httplib2.proxy_info_from_url(proxy_url, method=proxy_protocol)
+
+def get_new_http():
+  """Creates and returns a new httplib2.Http instance.
+  Returns:
+    An initialized httplib2.Http instance.
+  """
+
+  for proxy_env_var in ['http_proxy', 'https_proxy']:
```

Comment:
<!--new_thread; commit:1033d84efb2ada3802967025d2a64d01f3649c59; resolved:0-->
Please set `proxy_info = None` before this.